connexion/operations/openapi.py

```diff
@@ -268,36 +268,79 @@ def body_definition(self):
+
+        # if the body came in null, and the schema says it can be null, we decide
+        # to include no value for the body argument, rather than the default body
         if is_nullable(self.body_schema) and is_null(body):
-            return {x_body_name: None}
+            if x_body_name in arguments or has_kwargs:
+                return {x_body_name: None}
+            return {}
 
+        # now determine the actual value for the body (whether it came in or is default)
         default_body = self.body_schema.get('default', {})
-        body_props = {k: {"schema": v} for k, v
+        body_props = {sanitize(k): {"schema": v} for k, v
                       in self.body_schema.get("properties", {}).items()}
 
-        # by OpenAPI specification `additionalProperties` defaults to `true`
-        # see: https://github.com/OAI/OpenAPI-Specification/blame/3.0.2/versions/3.0.2.md#L2305
-        additional_props = self.body_schema.get("additionalProperties", True)
-
         if body is None:
             body = deepcopy(default_body)
 
+        # if the body isn't even an object, then none of the concerns below matter
         if self.body_schema.get("type") != "object":
             if x_body_name in arguments or has_kwargs:
                 return {x_body_name: body}
             return {}
 
-        body_arg = deepcopy(default_body)
-        body_arg.update(body or {})
+        # by OpenAPI specification `additionalProperties` defaults to `true`
+        # see: https://github.com/OAI/OpenAPI-Specification/blame/3.0.2/versions/3.0.2.md#L2305
+        additional_props = self.body_schema.get("additionalProperties", True)
 
-        res = {}
-        if body_props or additional_props:
-            res = self._get_typed_body_values(body_arg, body_props, additional_props)
+        # supply the initial defaults and convert all values to the proper types by schema
+        x = deepcopy(default_body)
+        x.update(body or {})
+        converted_body = self._get_typed_body_values(x, body_props, additional_props)
+
+        # NOTE:
+        # Handlers could have a single argument to receive the whole body or they
+        # could have individual arguments to receive all the body's parameters, or
+        # they may have a **kwargs, arguments to receive anything.  So, the question
+        # arises that if kwargs is given, do we pass to the handler a single body
+        # argument, or the broken out arguments, or both?
+        #
+        # #1 If 'x-body-arg' is explicitly given and it exists in [arguments], then the
+        # body, as a whole, will be passed to the handler with that name. STOP.
+        #
+        # #2 If kwargs is given, then we don't know what the handler cares about, so we
+        # pass the body as a whole as an argument named, 'body', along with the
+        # individual body properties. STOP.
+        #
+        # #3 Else, we pass the body's individual properties which  exist in [arguments].
```

Comment:
I don't think this piece makes sense to add. IIRC this was the behavior for swagger2, but I think it's an anti-pattern. Passing in a body object avoids some issues like parameter name conflicts, and is much less "magical".

Reply:
It'd sure break a lot of code to remove #3.  It's the whole "Automatic Parameter Handling" that's mentioned in the examples in README.rst.  This PR is only preserving that behavior, not changing it.

Reply:
Agree with @dtkav here. This isn't the current behavior for OpenAPI 3 and I don't think we should support it.

Reply:
@RobbeSneyders, thanks for taking the time to look at this.

I just wanted to get clarification on this.  So you're saying that if there's a body parameter named 'x' declared in the spec and there's an argument named 'x' in the handler, then it should not try to match that body parameter to the handler argument?  Isn't that inconsistent with how query and path parameters work?  Is there a reason they should differ in behavior?  I'm guessing what @dtkav was getting at was that body params may conflict with the others?  But couldn't they conflict with each other just the same?  Is there a conflict resolution strategy for path and query?

If the fear is breaking existing apps, that's understandable.  My thought was that dealing with body directly is breaking the abstraction that connexion is trying to put over those lower level details.

Again, I'm willing to change it.. just trying to understanding the reasoning for future reference.

Thanks

Reply:
> So you're saying that if there's a body parameter named 'x' declared in the spec and there's an argument named 'x' in the handler, then it should not try to match that body parameter to the handler argument?

Indeed, we should only pass the body as a whole. I don't think this is inconsistent with the query and path parameters. Query and path parameters are top level, just as the body. The content of the body is one level below. We only pass the top level.

Next to this, passing the body is a lot less magical as @dtkav mentioned. The user knows what to expect, which isn't the case with the current implementation in this PR, shown by the 5 paragraph comment needed to explain the behavior :slightly_smiling_face:

Path and query parameters can indeed also conflict with each other. Currently [the query parameters overwrite the path parameters](https://github.com/spec-first/connexion/blob/main/connexion/operations/abstract.py#L280). Probably not ideal, however conflicting parameters should ideally be avoided by the user.

Reply:
So, I reverted back to the original behavior for _get_body_argument() which no longer breaks body parameters into handler arguments.

However, in the process I learned, or.. remembered (this original change being so long ago now).. that the reason I though body should be split out is because uploaded files in the body _are_ broken out as individual handler arguments 
 which is an existing behavior apparently.

My main change here was to make the aiohttp handler consistent with flask.  I did that by constructing the ConnexionRequest() object in aiohttp_api.py the same as flask does regarding files.

So, I'm just pointing out that some future cleanup might want to make the file elements stay inside of body, where I'd think you'd say they belong due to the reasoning that they're not top-level parameters.  But I think it doesn't work that way because this is basically flask's existing behavior.

Reply:
I think we're ready to merge now.  Thanks!